test/e2e/tests/delete-approved-tx.spec.js

```diff
@@ -0,0 +1,44 @@
+const { strict: assert } = require('assert');
+const { convertToHexValue, withFixtures } = require('../helpers');
+
+describe('Failed approved but not submitted transaction on boot', function () {
+  const ganacheOptions = {
+    accounts: [
+      {
+        secretKey:
+          '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
+        balance: convertToHexValue(25000000000000000000),
+      },
+    ],
+  };
+  it('should set the approved but not submitted transaction to failed', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: 'approved-tx',
+        ganacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.clickElement({
+          text: 'Activity',
+          tag: 'button',
+        });
+
+        const completedTransactions = await driver.findElements(
+          '.transaction-list__completed-transactions',
+        );
+        assert.equal(completedTransactions.length, 1);
+        const transactionStatus = await driver.findElements(
+          '.transaction-status',
+        );
+
+        assert.equal(await transactionStatus.length, 1);
```

Comment:
Does this assertion guarantee that the status is now `failed`? Should we drill into the value of the `transaction-status` element?

Reply:
fixed [here](https://github.com/MetaMask/metamask-extension/pull/13360/commits/65f822c84e491070b2ef5a0114256eb921d65aad)